src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -77,15 +96,23 @@ protected override void OnNavigatedTo(NavigationEventArgs e)
                     {
                         _ = _queue.EnqueueAsync(() =>
                         {
+                            var result = (bool)lvm.InitializeCommand.ExecutionTask.GetResultOrDefault()!;
+
                             ViewModel = lvm;
-                            LoadedState = ViewModelLoadedState.Loaded;
+                            WeakReferenceMessenger.Default.Send<UpdateActionBarPage>(new(lvm));
+                            LoadedState = result ? ViewModelLoadedState.Loaded : ViewModelLoadedState.Error;
+                            if (!result)
+                            {
+                                WeakReferenceMessenger.Default.Send<UpdateActionBarMessage>(new(null));
+                            }
```

Comment:
```suggestion
                            WeakReferenceMessenger.Default.Send<UpdateActionBarPage>(new(result ? lvm : null));
                            LoadedState = result ? ViewModelLoadedState.Loaded : ViewModelLoadedState.Error;
```
Could this not be simplified?

Reply:
yes, yes it can

Reply:
Yup, totally valid, just with everything else with messaging, we just need to be cognizant of the lifetimes of objects and registration/deregistration.